Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,511 @@
+    {
+        if (m_tabWidget->currentIndex() == 0)
+        {
+            bool templateLocationIsValid = ValidateGemTemplateLocation();
+            if (!m_formFolderRadioButton->isChecked())
+            {
+                m_backButton->setVisible(true);
+                m_tabWidget->setCurrentIndex(1);
+            }
+            else if (templateLocationIsValid && m_formFolderRadioButton->isChecked())
+            {
+                m_backButton->setVisible(true);
+                m_tabWidget->setCurrentIndex(1);
+            }
+        }
+        else if (m_tabWidget->currentIndex() == 1)
+        {
+            bool displayNameIsValid = ValidateGemDisplayName();
+            bool systemNameIsValid = ValidateGemSystemName();
+            bool licenseNameIsValid = ValidateLicenseName();
+            bool globalGemTagIsValid = ValidateGlobalGemTag();
+            bool optionalGemTagsAreValid = ValidateOptionalGemTags();
+            if (systemNameIsValid && licenseNameIsValid && globalGemTagIsValid && displayNameIsValid && optionalGemTagsAreValid)
+            {
+                m_createAGemInfo.m_displayName = m_gemDisplayName->lineEdit()->text();
+                m_createAGemInfo.m_name = m_gemSystemName->lineEdit()->text();
+                m_createAGemInfo.m_summary = m_gemSummary->lineEdit()->text();
+                m_createAGemInfo.m_requirement = m_requirements->lineEdit()->text();
+                m_createAGemInfo.m_licenseText = m_license->lineEdit()->text();
+                m_createAGemInfo.m_licenseLink = m_licenseURL->lineEdit()->text();
+                m_createAGemInfo.m_licenseText = m_license->lineEdit()->text();
+                m_createAGemInfo.m_documentationLink = m_documentationURL->lineEdit()->text();
+                m_createAGemInfo.m_path = m_gemLocation->lineEdit()->text();
+
+                m_tabWidget->setCurrentIndex(2);
+                m_nextButton->setText(tr("Create"));
+
+            }
+        }
+        else if (m_tabWidget->currentIndex() == 2)
+        {
+            bool creatorNameIsValid = ValidateCreatorName();
+            bool repoURLIsValid = ValidateRepositoryURL();
+            if (creatorNameIsValid && repoURLIsValid)
+            {
+                m_createAGemInfo.m_creator = m_creatorName->lineEdit()->text();
+                m_createAGemInfo.m_repoUri = m_repositoryURL->lineEdit()->text();
+                auto result = PythonBindingsInterface::Get()->CreateGem(m_gemTemplateLocation->lineEdit()->text(), m_createAGemInfo);
+                emit CreateButtonPressed();
+            }
+        }
+    }
+
+    void CreateAGemScreen::UpdateNextButtonToCreate()
+    {
+        if (m_tabWidget->currentIndex() == 2)
+        {
+            m_nextButton->setText(tr("Create"));
+        }
+        else
+        {
+            m_nextButton->setText(tr("Next"));
+        }
+    }
+
+    void CreateAGemScreen::AddDropdownActions(FormComboBoxWidget* dropdown)
```

Comment:
:grey_question: these are supposed to be dynamically generated right? Pretty sure we don't want to include a number of these like "Slices","Synergy","Barrier","Unity" etc...

Reply:
Yes, I'll be updating this to do something similar to what gems do (generates the list based on all the labels it's encountered)